File: DataAccess/Entities/Vehicle/VehicleColor.cs
namespace DataAccess.Entities;

public class VehicleColor
{
    public int ID;

    public Vehicle? Vehicle;
    public int VehicleID;

    public required string Name;
    public required string HexCode;

    public required bool IsPublished;

    public required bool IsDeleted;
    public DateTime? DeletedAt;
}

File: Frontend/website/src/classes/[user]/vehicle-color.ts
import type { tVehicleColorModel } from "@/models/vehicle-color";

class clsVehicleColor {
  public uuid: string;

  public name: string;
  public hexCode: string;

  public constructor(vehicleColor: tVehicleColorModel) {
    this.uuid = vehicleColor.uuid;

    this.name = vehicleColor.name;
    this.hexCode = vehicleColor.hexCode;
  }
}

export { clsVehicleColor };
